hphp/runtime/vm/jit/vasm-reuseimm.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+   +----------------------------------------------------------------------+
+   | HipHop for PHP                                                       |
+   +----------------------------------------------------------------------+
+   | Copyright (c) 2010-present Facebook, Inc. (http://www.facebook.com)  |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.php.net/license/3_01.txt                                  |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+*/
+
+#include <cstdlib>
+
+#include "hphp/runtime/vm/jit/vasm.h"
+
+#include "hphp/runtime/vm/jit/vasm-gen.h"
+#include "hphp/runtime/vm/jit/vasm-instr.h"
+#include "hphp/runtime/vm/jit/vasm-print.h"
+#include "hphp/runtime/vm/jit/vasm-reg.h"
+#include "hphp/runtime/vm/jit/vasm-unit.h"
+#include "hphp/runtime/vm/jit/vasm-util.h"
+
+#include "hphp/util/assertions.h"
+
+TRACE_SET_MOD(vasm);
+
+namespace HPHP { namespace jit {
+
+struct qTrack {
+  Immed64 val;
+  Vreg base;
+  int  ttl;
+};
+
+struct Env {
+  Vunit& unit;
+  int tick;
+  std::deque<qTrack> qdeque; // leaky bucket of recent immediates
+};
+
+static bool isMultiword(int64_t imm) {
```

Comment:
Everything in this file (notably including the class definitions) except the one exposed function should be enclosed in an anonymous namespace.  Then you can strip all these `static`s.

Also: This `isMultiword()` predicate seems fairly ARM-specific, but this file is purportedly architecture-independent.